app/models/resource_action_workflow.rb

```diff
@@ -29,10 +29,18 @@ def submit_request
     return result unless result[:errors].blank?
 
     values = create_values_hash
+
+    service_order = ServiceOrder.create_service_order('generated_service_order', 'cart', @requester)
```

Comment:
The service order name should be "Order # ID" (where ID is the `id` of the service_order object).

I am thinking you should do that as part of an after_create method on ServiceOrder, but maybe @Fryguy knows a better way.  Using after_create also requires a change to `validates :name`.

```ruby
  validates :name, :presence => true, :on => :update

  after_create  :create_order_name

  def create_order_name
    update_attributes(:name => "Order # #{id}") if name.blank?
  end
```